### PR TITLE
Fixing searching users without team

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing searching users without team

#### Ticket Link
[MM-23610](https://mattermost.atlassian.net/browse/MM-23610)

### CODE REVIEW

File: store/searchlayer/user_layer.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ func (s *SearchUserStore) Search(teamId, term string, options *model.UserSearchO
 				mlog.Error("Encountered error on Search.", mlog.String("search_engine", engine.GetName()), mlog.Err(err))
 				continue
 			}
-			if len(listOfAllowedChannels) == 0 {
+
```
Comment (ethervoid) [inline store/searchlayer/user_layer.go]: What was the root cause? I don't have enough context right now. Thank you!

Comment (jespino) [reply store/searchlayer/user_layer.go]: The idea is, listOfAllowedChannels can be 0 length if you have no channels allowed there, or can be nil if you have no restrictions about what channels you can see.

**Comment by jespino** [comment]
@ethervoid it is not easy yet to test properly ElasticSearch enable use cases and elasticsearch disable use cases, probably makes sense to wait until the elasticsearch test suite is ready and then add the test case there.

**Comment by amyblais** [comment]
@jespino / @mgdelacroix Please help merge + cherry-pick.

**Pull request merged**